server/src/test/java/org/opensearch/extensions/ExtensionsManagerTests.java

```diff
@@ -528,6 +529,86 @@ public void testHandleExtensionRequest() throws Exception {
         assertEquals("Handler not present for the provided request", exception.getMessage());
     }
 
+    public void testExtensionRequest() throws Exception {
+        ExtensionsManager.RequestType expectedRequestType = ExtensionsManager.RequestType.REQUEST_EXTENSION_DEPENDENCY_INFORMATION;
+
+        // Test ExtensionRequest 2 arg constructor
+        Optional<String> expectedUniqueId = Optional.of("test uniqueid");
+        ExtensionRequest extensionRequest = new ExtensionRequest(expectedRequestType, expectedUniqueId.orElse(null));
```

Comment:
This works but I think it would be more readable and conform to the method signature as:
```java
String expectedUniqueId = "test uniqueid";
ExtensionRequest extensionRequest = new ExtensionRequest(expectedRequestType, expectedUniqueId);
// and when tested on lines 539 and 548...
assertEquals(Optional.of(expectedUniqueID), extensionRequest.getUniqueId());
```